test/controllers/works_controller_test.rb

```diff
@@ -0,0 +1,133 @@
+      Work.destroy_all
+      get works_path
+      must_respond_with :success
+    end
+  end
+
+  describe "new" do
+    it "returns a success" do
+      get new_work_path
+      must_respond_with :success
+    end
+  end
+
+  describe "create" do
+    it "redirects to works_path when work data is valid" do
+      work_data = {
+        work: {
+          title: "test",
+          creator: "test",
+          category: "album"
+        }
+      }
+      Work.new(work_data[:work]).must_be :valid?
+
+      start_work_count = Work.count
+
+      post works_path, params: work_data
+      must_respond_with :redirect
+      must_redirect_to works_path
+
+      Work.count.must_equal start_work_count + 1
+    end
+    it "sends bad request when work data is invalid" do
+      invalid_work_data = {
+        work: {
+          title: "",
+          creator: "",
+          category: ""
+        }
+      }
+      Work.new(invalid_work_data[:work]).wont_be :valid?
+
+      start_work_count = Work.count
+
+      post works_path, params: invalid_work_data
+      must_respond_with :bad_request
+
+      Work.count.must_equal start_work_count
+    end
+  end
+
+  describe "show" do
+    it "returns sucess when given a valid work id" do
+      work_id = Work.first.id
+      get work_path(work_id)
+      must_respond_with :success
+    end
+    it "returns not found when given a invalid work id" do
+      work_id = Work.first.id + 1
+      get work_path(work_id)
+      must_respond_with :not_found
+    end
+  end
+
+  describe "edit" do
+    it "returns success when given a valid work id" do
+      work_id = Work.first.id
+      get edit_work_path(work_id)
+      must_respond_with :success
+    end
+    it "returns not found when given a valid work id" do
+      work_id = Work.first.id + 1
+      get edit_work_path(work_id)
+      must_respond_with :not_found
+    end
+  end
+
+  describe "update" do
+    it "returns success if work id is valid and change is valid" do
+      work = Work.first
+      work_data = {
+        work: {
+          title: "a different title",
+          creator: "happy",
+          category: "album"
+        }
+      }
+      work.update_attributes(work_data[:work])
+      work.must_be :valid?
+
+      patch work_path(work), params: work_data
+
+      must_respond_with :redirect
+      must_redirect_to work_path(work)
+
+    end
+
+    it "returns not found if the work id is invalid" do
+      work_id = Work.first.id + 1
+      get work_path(work_id)
+      must_respond_with :not_found
+    end
+    it "returns bad request if the change is invalid" do
+
+    end
+  end
+
+  ## check if this is valid?
+  describe "destroy" do
+    it "exist" do
+      work_id = Work.first.id
+      get work_path(work_id)
+      must_respond_with :success
+    end
+    ## currently errors
+    it "does not exist" do
+      work_id = Work.first.id
+      delete work_path(work_id)
+      must_respond_with :redirect
+    end
```

Comment:
What happens if you attempt to destroy a work that has votes? Can you do it? What happens to those votes?